### ISSUE TITLE
Als Squeak-Entwickler möchte ich, dass Profilingergebnisse gespeichert und wieder betrachtet werden können, damit ich zu einem späteren Zeitpunkt darauf zugreifen und die Werte mit aktuellen Vergleichen kann.

### ISSUE THREAD

**Issue by linusha** [hpi-swa-teaching/InteractiveProfilingTool#12]

**Comment by linusha** [comment]
Estimate: 6h

**Comment by tillprochaska** [comment]
I have no idea how to persist data within the image and couldn’t find any docs about it. Any ideas?

**Comment by linusha** [comment]
> I have no idea how to persist data within the image and couldn’t find any docs about it. Any ideas? Also:

This is probably not the nicest way to do this, but as far as my understanding of Squeak goes: Couldn't we have a List of Profiling Results as a Class Variable for the Model? I was under the impression that we could just keep it as an Object and Squeak would handle Serialization,... however this works with everything else in the Image if one presses save.

**Comment by twollnik** [comment]
I like linus's idea of saving the profiling results in the image and not writing them to disk ourselves. It may make sense to introduce a class `IPTResultsSaver` to encapsulate the storage and accessing logic.

**Comment by linusha** [comment]
> I like linus's idea of saving the profiling results in the image and not writing them to disk ourselves. It may make sense to introduce a class `IPTResultsSaver` to encapsulate the storage and accessing logic.

Agreed! :+1: 
Since everything we do in Squeak is encapsulated anyhow _and with the current project nobody needs anything else besides squeak anyways,_ I would strongly recommend that we try to avoid everything that would entail really writing something to disc manually.

**Comment by tillprochaska** [comment]
My point wasn’t about writing to the file system manually, but researching the goto method to persist user data in the image in Squeak. Using class variables has the disadvantage that users will loose all their data when they update the package.

**Comment by linusha** [comment]
> My point wasn’t about writing to the file system manually, but researching the goto method to persist user data in the image in Squeak. Using class variables has the disadvantage that users will loose all their data when they update the package.

True, sorry for the confusion. I am strongly in favor of talking about the specifics of this with our customers if we are to implement this, however:
We can write a complete tree of Tallys on disk using a ```ReferenceStream``` (or maybe even a ```SmartRefStream``` from the ```System-Object Storage``` Package.

**Comment by twollnik** [comment]
We opted for not deciding on an estimate today. Instead we want to talk to patrick / the mailing list and find out what options we have for persisting data in the image.
